nemo/utils/callbacks/nemo_model_checkpoint.py

```diff
@@ -158,24 +159,17 @@ def on_save_checkpoint(self, trainer, pl_module, checkpoint):
 
         if self.save_best_model:
             if not os.path.exists(maybe_injected_best_model_path):
-                return
+                return output
 
             if self.best_model_path == self.previous_best_path:
                 logging.debug('Best model has not changed, skipping save.')
                 return output
 
             self.previous_best_path = self.best_model_path
-            old_state_dict = deepcopy(pl_module.state_dict())
-            checkpoint = torch.load(maybe_injected_best_model_path, map_location='cpu')
-            if 'state_dict' in checkpoint:
-                checkpoint = checkpoint['state_dict']
-            # get a new instanace of the model
-            pl_module.load_state_dict(checkpoint, strict=True)
```

Comment:
@itzsimpl curious why L168-L173 and L178 are being removed ? Are they redundant ?

Reply:
By my understanding, correct me if I'm wrong, their function is to reload the best checkpoint, to have it in memory for saving the .nemo archive. However, with the addition of the `previous_best_model` variable and skipping saves when `best_model == previous_best_model`, the change can come only from the current checkpoint, the one that is currently in memory (assuming there are no external processes injecting better checkpoints). This makes reloading "self" into memory redundant.